File: frontend/src/merchant/object/application/MerchantRegistrationRequest.ts
import type {
  DisplayText,
  ImageUrl,
  Minutes,
  NoteText,
  PersonName,
  StoreCategory,
} from '@/shared/object/domain/DomainObjects'
import type { BusinessHours } from '@/merchant/object/store/BusinessHours'

export type MerchantRegistrationRequest = {
  merchantName: PersonName
  storeName: DisplayText
  category: StoreCategory
  businessHours: BusinessHours
  avgPrepMinutes: Minutes
  imageUrl?: ImageUrl
  note?: NoteText
}

File: backend/src/main/scala/merchant/object/application/MerchantRegistrationRequest.scala
package domain.merchant

import domain.shared.given

import domain.shared.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class MerchantRegistrationRequest(
    merchantName: PersonName,
    storeName: DisplayText,
    category: DisplayText,
    storeAddress: AddressText,
    location: Option[StoreLocation],
    businessHours: BusinessHours,
    avgPrepMinutes: Minutes,
    imageUrl: Option[ImageUrl],
    note: Option[NoteText],
)
object MerchantRegistrationRequest:
  given Encoder[MerchantRegistrationRequest] = deriveEncoder
  given Decoder[MerchantRegistrationRequest] = deriveDecoder
